usaspending_api/search/delta_models/award_search.py

```diff
@@ -291,8 +291,16 @@
   COALESCE(transaction_fpds.legal_entity_county_name, transaction_fabs.legal_entity_county_name) AS recipient_location_county_name,
   LPAD(CAST(CAST(REGEXP_EXTRACT(COALESCE(transaction_fpds.legal_entity_congressional, transaction_fabs.legal_entity_congressional), '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 2, '0')
             AS recipient_location_congressional_code,
-  -- TODO: include recipient_location_congressional_code_current derivation
-  'TEST CUR REC CONGR AS' AS recipient_location_congressional_code_current,
+  (CASE
+      WHEN (
+          COALESCE(transaction_fpds.legal_entity_country_code, transaction_fabs.legal_entity_country_code) <> 'USA'
+      ) THEN NULL
+      WHEN (
+          COALESCE(transaction_fpds.legal_entity_country_code, transaction_fabs.legal_entity_country_code) = 'USA'
+          AND COALESCE(rl_state_cd.congressional_district_no, rl_zips.congressional_district_no, rl_cd_zips_grouped.congressional_district_no, rl_cd_city_grouped.congressional_district_no, rl_cd_county_grouped.congressional_district_no) IS NOT NULL
```

Comment:
This part of the check might be unnecessary too. Couldn't you use the output value of the `COALESCE` whether it's null or not?

Reply:
Oh I see. Just do `ELSE` with the `COALESCE`. If it's `NULL` it hits the old `ELSE`. If not, it hits the old `WHEN`.